### DISCUSSION TITLE
How to get current locale information in Nextjs 13 app router without using useRouter hook?

### DISCUSSION THREAD

**Discussion by huskyjp** [vercel/next.js#54183] [Help]
### Summary

I have been using nextjs page structure for a while for multi-language website. I started to migrate to the app router then noticed that `useRouter` hook has been made slight change. (It removed `locale` property). 

I have read [the official doc here](https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration#step-5-migrating-routing-hooks:~:text=The%20locale%2C%20locales%2C%20defaultLocales%2C%20domainLocales%20values%20have%20been%20removed%20because%20built%2Din%20i18n%20Next.js%20features%20are%20no%20longer%20necessary%20in%20the%20app%20directory.%20Learn%20more%20about%20i18n.) but couldn't understand how to retrieve locale information in app router structure. 

My usecase is that I want to switch component between locales so I want to access to the current locale information from hook - how to achieve this?

### Additional information

```js
I have the following project folder structure.


- app
  - page.tsx
- components
  - Home.tsx
- hooks
  - useLocale.ts
- pages
  - index.tsx
- styles
```

In `useLocale.ts`, I simply retrieve the locale information from useRouter hook. (Currently not working)
```
import { useRouter } from "next/router"

export const useLocale = () => {
  const {locale} = useRouter()
  const isEnglish = locale == "en" ? true : false

  return { isEnglish }
}
```

In `index.tsx`, I call useLocale hook to choose which component/text to load. 
```
import type { NextPage } from "next"

import { Home } from "@components/Home"

const Index: NextPage = () => {
  const { isEnglish } = useLocale()

  return <Home exampleText={isEnglish? "Hello" : "Hola"} />
}

export default Index
```

**Comment by icyJoseph** [comment]
The new App Router doesn't come from `next/router`, it comes from `next/navigation`, but even that one doesn't have a locale property, it is only routing methods nowadays.

According to https://nextjs.org/docs/app/building-your-application/routing/internationalization, you have to use the `lang`uage as part of the path, and access it through `params`, which are passed to the Page component, or by using the `useParams` hook from `next/navigation`, https://nextjs.org/docs/app/api-reference/functions/use-params
